Database/compiler.lua

```diff
@@ -1186,13 +1277,15 @@ function QuestieDBCompiler:GetDBHandle(data, pointers, skipMap, keyToRootIndex,
     handle.stream = stream
 
     if overrides then
+        ---@param id QuestId|ObjectId|ItemId|NpcId
+        ---@param key string
+        ---@return any
         handle.QuerySingle = function(id, key)
```

Comment:
Inline function creation is bad, especially if the function it is created in is called multiple times. Can we extract that function together with all the other `handle.X = function` things below?

Reply:
Well maybe, but i think that it is outside of the scope of this PR, this is the way Aero did it way back and i just added some Emmy for it and some performance changes.